apps/src/aichat/views/ModelCustomizationWorkspace.tsx

```diff
@@ -1,19 +1,24 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {useSelector} from 'react-redux';
 
 import {AichatLevelProperties} from '@cdo/apps/aichat/types';
+import {FontAwesomeV6IconProps} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import Tabs, {TabsProps} from '@cdo/apps/componentLibrary/tabs/Tabs';
 import {isReadOnlyWorkspace} from '@cdo/apps/lab2/lab2Redux';
 import {useAppSelector} from '@cdo/apps/util/reduxHooks';
 
 import PublishNotes from './modelCustomization/PublishNotes';
 import RetrievalCustomization from './modelCustomization/RetrievalCustomization';
 import SetupCustomization from './modelCustomization/SetupCustomization';
 import {isVisible, isDisabled} from './modelCustomization/utils';
-import Tabs, {Tab} from './tabs/Tabs';
 
 import styles from './model-customization-workspace.module.scss';
 
 const ModelCustomizationWorkspace: React.FunctionComponent = () => {
+  const [setupTabName, setSetupTabName] = useState('Setup');
```

Comment:
Potential optimization: I think we might be able to use `useMemo` to simplify this a bit, and not have to store all these state fields separately (haven't actually tested this though so I might be missing something). Ex:
```
const [selectedTab, setSelectedTab] = useState('setup');

const visibleTabs = useMemo(() => {
  if (showSetupCustomization) {
    visibleTabs.push({
      value: 'setup',
      text: "Setup" + (isSetupCustomizationReadonly && selectedTab === 'setup' ? ' (view only') : '');
      ...other fields
      });
  }
  ... same for the other tabs

}, [selectedTab, ... other dependencies]);
```
The idea is to cache the list of `visibleTabs` with a `useMemo` that recalculates whenever `selectedTab` updates. The `onChange` handler provided to the `Tabs` component should update `selectedTab` which should automatically cause `visibleTabs` to be recalculated, and passed as new props to the `Tabs` component to show the new labels. We'll want to make sure to wrap the `handleOnChange` in a `useCallback` to prevent additional re-renders too.

Reply:
Sure! Will update. I included the assignment of the icon within the `useMemo as well`.